x-pack/legacy/plugins/file_upload/public/components/json_index_file_picker.js

```diff
@@ -21,117 +21,114 @@ export class JsonIndexFilePicker extends Component {
-              id="xpack.fileUpload.jsonIndexFilePicker.acceptableTypesError"
-              defaultMessage="File is not one of acceptable types: {types}"
-              values={{ types }}
-            />
-          )
-        });
+    if (fileArr.length === 0) { // Remove
+      return;
+    } else if (fileArr.length === 1) { // Parse & index file
+      const file = fileArr[0];
+      if (!this._validateFile(file)) {
         return;
       }
-      const initIndexName = splitNameArr[0];
-      setIndexName(initIndexName);
+      await this._parseFile(file);
+    } else {
+      // No else
+    }
+  };
 
-      // Check valid size
-      const { size } = file;
-      if (size > MAX_FILE_SIZE) {
-        this.setState({
-          fileUploadError: (
-            <FormattedMessage
-              id="xpack.fileUpload.jsonIndexFilePicker.acceptableFileSize"
-              defaultMessage="File size {fileSize} bytes exceeds max file size of {maxFileSize}"
-              values={{
-                fileSize: size,
-                maxFileSize: MAX_FILE_SIZE
-              }}
-            />
-          )
-        });
-        return;
-      }
+  _validateFile(file) {
+    const { resetFileAndIndexSettings, setIndexName } = this.props;
+    if (!file.name) {
+      this.setState({
+        fileUploadError: i18n.translate(
+          'xpack.fileUpload.jsonIndexFilePicker.noFileNameError',
+          { defaultMessage: 'No file name provided' })
+      });
+      return false;
+    }
 
-      // Parse file
-      this.setState({ fileParsingProgress: i18n.translate(
-        'xpack.fileUpload.jsonIndexFilePicker.parsingFile',
-        { defaultMessage: 'Parsing file...' })
+    // Check file type, assign default index name
+    const splitNameArr = file.name.split('.');
+    const fileType = splitNameArr.pop();
+    const types = ACCEPTABLE_FILETYPES.reduce((accu, type) => {
+      accu = accu ? `${accu}, ${type}` : type;
+      return accu;
+    }, '');
+    if (!ACCEPTABLE_FILETYPES.includes(fileType)) {
+      this.setState({
+        fileUploadError: (
+          <FormattedMessage
+            id="xpack.fileUpload.jsonIndexFilePicker.acceptableTypesError"
+            defaultMessage="File is not one of acceptable types: {types}"
+            values={{ types }}
+          />
+        )
       });
-      const parsedFileResult = await parseFile(
-        file, onFileUpload, transformDetails
-      ).catch(err => {
-        this.setState({
-          fileUploadError: (
-            <FormattedMessage
-              id="xpack.fileUpload.jsonIndexFilePicker.unableParseFile"
-              defaultMessage="Unable to parse file: {error}"
-              values={{
-                error: err.message
-              }}
-            />
-          )
-        });
+      resetFileAndIndexSettings();
+      return false;
+    }
+    const initIndexName = splitNameArr[0];
+    setIndexName(initIndexName);
+
+    // Check valid size
+    const { size } = file;
```

Comment:
similar here. what does this add?

This pattern is useful in function params imho, because it's JS way of having named parameters. It _may_ be useful if the variable is reused often later on in the function. But this is basically just equivalent to `const size = file.size`, which seems unnecessary.